### DISCUSSION TITLE
Custom NodeJS code?

### DISCUSSION THREAD

**Discussion by finom** [vercel/next.js#20504] [Help]
Hey guys, I'm a complete noob at next.js. I've read mostly all of the documentation but haven't figured out how to run custom NodeJS code there. My goal is to run a `setInterval` once server is run but the task isn't covered by common next.js patterns: it isn't a custom server (I'm going to use Vercel), it isn't an API endpoint, it isn't a rendered JSX... Can you please tell me how to run something on next.js start? Thank you!

**Comment by jamesmosier** [comment]
The only way to run custom Node.js code alongside Next would be to use a [custom server](https://nextjs.org/docs/advanced-features/custom-server). Otherwise you'd need to run the Node script separately (i.e. `node my-custom-script.js && next start`). 

What are you planning on doing with this `setInterval`?

**Reply by finom** [reply]
Yeah, I've already used custom server. Unfortunately it required me to use Heroku instead of Vercel. 

> What are you planning on doing with this setInterval?

To ping an API and if there some specific data appears, then send SMS notifications to given set of people.

**Comment by lalit2005** [comment]
To run custom Node scripts, serverless functions can be used which can be hosted on Vercel alongside the frontend code.

**Comment by finom** [comment]
Wanted to make sure, does it mean that NextJS isn't compatible with any kind of real-time things if custom server isn't involved?

**Reply by lalit2005** [reply]
For Real time data that changes frequently, Server Side Rendering suits the best. Site is rendered on every request.
But SSR is not advised if the same result can be obtained in other ways such as `getStaticProps ` and `getStaticPaths `
